src/index.js

```diff
@@ -54,11 +54,22 @@ module.exports = class BlobbyClient extends EventEmitter {
   }
 
   getFile(storage, fileKey, opts) {
+    const $client = this;
     return new Promise((resolve, reject) => {
       storage.fetch(fileKey, opts, (err, headers, data) => {
-        if (err) return void reject(err);
-
-        resolve([headers, data]);
+        if (!err) return void resolve([headers, data]);
+        const { statusCode } = err;
+        const encodedFileKey = fileKey.split('/').map(encodeURIComponent).join('/');
+        if (encodedFileKey === fileKey || ![403, 404].includes(statusCode)) {
```

Comment:
I don't believe `statusCode` is currently being forwarded by drivers. Have you verified this works? I believe you'll also need to update the client drivers before this will behave as expected.

Reply:
Oh I see. It assumes `404` effectively if status is not set. Might be helpful to set line 61 to `statusCode = 404`

Reply:
yeah, I have a PR to expose the statusCode here. 
https://github.com/asilvas/blobby-s3/pull/3

Reply:
I was not aware of that 404 auto assumption logic, but I do agree to explicitly expose httpStatus code in the error from blobby-s3.